bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -698,7 +712,8 @@ void DWARFRewriter::updateDebugInfo() {
       DWODIEBuilder.updateDWONameCompDirForTypes(DWOStrOffstsWriter,
                                                  DWOStrWriter, **SplitCU,
                                                  DwarfOutputPath, DWOName);
-      DebugLoclistWriter DebugLocDWoWriter(*Unit, Unit->getVersion(), true);
+      DebugLoclistWriter DebugLocDWoWriter(*Unit, Unit->getVersion(), true,
+                                           *AddressWriter);
       DebugRangesSectionWriter *TempRangesSectionWriter = RangesSectionWriter;
       if (Unit->getVersion() >= 5) {
         TempRangesSectionWriter = RangeListsWritersByCU[*DWOId].get();
```

Comment:
You are not seem to be setting AddressWriter like you do for 
RangeListsSectionWriter->setAddressWriter(AddressWriter);
How is it working?

Reply:
It's set for RangeListWritersByCU when RangeListWritersByCU is populated, and TempRangesSectionWriter accesses RangeListsWritersByCU. We don't need to populate it in both places as far as I can tell.

Reply:
Ah I see line 654. Missed it. cool.